sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -0,0 +1,460 @@
+            self,
+            signed_identifiers,  # type: dict[str,AccessPolicy]
+            **kwargs):
+        # type: (...) -> None
+        """Sets stored access policies for the table that may be used with Shared Access Signatures.
+
+        :param signed_identifiers:
+        :type signed_identifiers: dict[str,AccessPolicy]
+        :return: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        self._validate_signed_identifiers(signed_identifiers)
+        identifiers = []
+        for key, value in signed_identifiers.items():
+            if value:
+                value.start = serialize_iso(value.start)
+                value.expiry = serialize_iso(value.expiry)
+            identifiers.append(SignedIdentifier(id=key, access_policy=value))
+        signed_identifiers = identifiers  # type: ignore
+        try:
+            self._client.table.set_access_policy(
+                table=self.table_name,
+                table_acl=signed_identifiers or None,
+                **kwargs)
+        except HttpResponseError as error:
+            process_table_error(error)
+
+    @distributed_trace
+    def create_table(
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> str
+        """Creates a new table under the current account.
+
+        :return: Table created
+        :rtype: str
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        table_properties = TableProperties(table_name=self.table_name, **kwargs)
+        table = self._client.table.create(table_properties)
+        return table.table_name
+
+    @distributed_trace
+    def delete_table(
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Deletes the table under the current account.
+
+        :return: None
+        :rtype: None
+        """
+        self._client.table.delete(table=self.table_name, **kwargs)
+
+    @distributed_trace
+    def delete_entity(
+            self,
+            partition_key,  # type: str
+            row_key,  # type: str
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Deletes the specified entity in a table.
+
+        :param partition_key: The partition key of the entity.
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+        :keyword str etag: Etag of the entity
+        :keyword ~azure.core.MatchConditions match_condition: MatchCondition
+        :return: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=kwargs.pop('etag', None),
+                                                                match_condition=kwargs.pop('match_condition', None)),
+                                                    etag_param='etag', match_param='match_condition')
```

Comment:
Can we open an issue to revisit the etag/match condition patterns? I know it came up in the arch-board discussion - and I don't want to change it now - but it would be good to have the issue open so we can remember to go back and clarify we got the design right here.